pkg/schedule/placement/rule.go

```diff
@@ -90,9 +90,7 @@ func (r *Rule) String() string {
 // Clone returns a copy of Rule.
 func (r *Rule) Clone() *Rule {
 	var clone Rule
-	json.Unmarshal([]byte(r.String()), &clone)
-	clone.StartKey = append(r.StartKey[:0:0], r.StartKey...)
-	clone.EndKey = append(r.EndKey[:0:0], r.EndKey...)
```

Comment:
After using `Unmarshal`, there's no need to reallocate the address.

Reply:
startKey is ``json:"-"`` which means cannot be Unmarshal

Reply:
Good catch!